src/notehead.ts

```diff
@@ -181,30 +229,33 @@ export class NoteHead extends Note {
   }
 
   // Get the `BoundingBox` for the `NoteHead`
-  getBoundingBox() {
+  getBoundingBox(): BoundingBox {
     if (!this.preFormatted) {
       throw new Vex.RERR('UnformattedNote', "Can't call getBoundingBox on an unformatted note.");
     }
+    if (!this.stave) throw new Vex.RERR('NoStave', "Can't call getBoundingBox without a stave.");
 
     const spacing = this.stave.getSpacingBetweenLines();
     const half_spacing = spacing / 2;
     const min_y = this.y - half_spacing;
 
-    return new Flow.BoundingBox(this.getAbsoluteX(), min_y, this.width, spacing);
+    return new BoundingBox(this.getAbsoluteX(), min_y, this.width, spacing);
   }
 
   // Set notehead to a provided `stave`
-  setStave(stave) {
+  setStave(stave?: Stave): this {
```

Comment:
Can you make this non-optional?

Reply:
Yes, with a small change in `StaveNote`.
Done